src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests.cs

```diff
@@ -42432,6 +42450,258 @@ static void Test(MyCollection a)
+                    public static IEnumerator<T> GetEnumerator<T>(this MyCollection<T>{{(allowNullReceiver ? "?" : "")}} c) => throw null!;
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollection<int>? x = null;
+                        object[] y = [..x];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            if (allowNullReceiver)
+            {
+                comp.VerifyEmitDiagnostics();
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (15,25): warning CS8604: Possible null reference argument for parameter 'c' in 'IEnumerator<int> Extensions.GetEnumerator<int>(MyCollection<int> c)'.
+                    //         object[] y = [..x];
+                    Diagnostic(ErrorCode.WRN_NullReferenceArgument, "x").WithArguments("c", "IEnumerator<int> Extensions.GetEnumerator<int>(MyCollection<int> c)").WithLocation(15, 25));
+            }
+        }
+
+        [Fact]
+        public void Spread_Nullable_05()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static IEnumerable<T> F<T>(T t)
+                    {
+                        return [t];
+                    }
+                    static void Main()
+                    {
+                        object x = null;
+                        var y = F(x);
+                        object[] z = [..y];
+                        z = [..y];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (11,20): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object x = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(11, 20),
+                // (13,25): warning CS8601: Possible null reference assignment.
+                //         object[] z = [..y];
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "y").WithLocation(13, 25),
+                // (14,16): warning CS8601: Possible null reference assignment.
+                //         z = [..y];
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "y").WithLocation(14, 16));
+        }
+
+        [Fact]
+        public void Spread_Nullable_06()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static void Main()
+                    {
+                        IEnumerable<string?> x = [null];
+                        IEnumerable<object> y = [..x];
```

Comment:
Consider adding the converse scenario, where the elements are not-null and the target collection allows for null elements:
```
                        IEnumerable<string> x = [""];
                        IEnumerable<object?> y = [..x]; // no warning
```